pagerduty/resource_pagerduty_service_integration.go

```diff
@@ -519,12 +519,21 @@ func flattenEmailParsers(v []*pagerduty.EmailParser) []map[string]interface{} {
 		var valueExtractors []map[string]interface{}
 
 		for _, ve := range ef.ValueExtractors {
-			extractor := map[string]interface{}{
-				"type":         ve.Type,
-				"value_name":   ve.ValueName,
-				"part":         ve.Part,
-				"starts_after": ve.StartsAfter,
-				"ends_before":  ve.EndsBefore,
+			if ve.Type == "regex" {
+				extractor := map[string]interface{}{
+					"type":       ve.Type,
+					"value_name": ve.ValueName,
+					"part":       ve.Part,
+					"regex":      ve.Regex,
+				}
+			} else {
+				extractor := map[string]interface{}{
+					"type":         ve.Type,
+					"value_name":   ve.ValueName,
+					"part":         ve.Part,
+					"starts_after": ve.StartsAfter,
+					"ends_before":  ve.EndsBefore,
+				}
```

Comment:
This is causing an `undefined: extractor` error. You should be able to fix it by adding `var extractor := map[string]interface{}` just before the for loop. And then change these `extractor` assignments to use just `=`.

Reply:
I've pushed the fix and verified that provider compiles on my local machine.
Faced issues when tried to run acceptance test, so wasn't able to verify changes in it (yet it's very minor).